tiledb/sm/fragment/fragment_metadata.cc

```diff
@@ -530,6 +530,36 @@ URI FragmentMetadata::var_uri(const std::string& name) const {
   return fragment_uri_.join_path(name + "_var" + constants::file_suffix);
 }
 
+Status FragmentMetadata::load_tile_offsets(
+    const EncryptionKey& encryption_key, std::vector<std::string>&& names) {
+  // Sort 'names' in ascending order of their index. The
+  // motivation is to load the offsets in order of their
+  // layout for sequential reads to the file.
+  std::sort(
+      names.begin(),
+      names.end(),
+      [&](const std::string& lhs, const std::string& rhs) {
+        assert(idx_map_.count(lhs) > 0);
+        assert(idx_map_.count(rhs) > 0);
+        return idx_map_[lhs] < idx_map_[rhs];
+      });
+
+  // The fixed offsets are located before the
+  // var offsets. Load all of the fixed offsets
+  // first.
+  for (const auto& name : names) {
```

Comment:
I wonder whether we can give a hint to to the read-ahead-cache here. For large datasets, the tile offsets can easily be in the order of 100s of MB. Otherwise, perhaps we should parallelize these loops.

Reply:
I see below that function `load_tile_offsets` is parallelized within an attribute, whereas this for loop is across attributes. Parallelizing this for may still have gains when we are reading from numerous attributes.

Reply:
I agree that we can optimize here. There are a few different things we could do. I will create a ticket to track this. You are right that we want to parallelize the reads, but we also have to be careful of the VFS read-ahead. If we launch all read tasks in parallel, we can't guarantee the reads will be sequential and then we'll regress in performance.

So I think the right thing to do is to coalesce the reads and parallelize reads to the coalesced ranges. We would need some sort of "fudge factor" when coalescing because there may be scenarios where it makes sense to coalesce non-contiguous reads to reduce the total number of reads. So: lots of tuning to figure out what exactly that "fudge factor" is.

This would be helpful for scenarios where `fragments * number of read attributes` is less than the total number of cpu cores, so we definitely want to do this at some point.